File: SMarketPlace/SMarketPlace.Web/Modules/ServerTypes/Ecommerce/OrdenesEcommerceRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface OrdenesEcommerceRow {
    Consecutivo?: number;
    IdPedidoEcommerce?: string;
    EstadoPedidoInternet?: string;
}

export abstract class OrdenesEcommerceRow {
    static readonly idProperty = 'Consecutivo';
    static readonly nameProperty = 'IdPedidoEcommerce';
    static readonly localTextPrefix = 'Ecommerce.OrdenesEcommerce';
    static readonly lookupKey = 'Ecommerce.OrdenesEcommerce';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<OrdenesEcommerceRow>('Ecommerce.OrdenesEcommerce') }
    static async getLookupAsync() { return getLookupAsync<OrdenesEcommerceRow>('Ecommerce.OrdenesEcommerce') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<OrdenesEcommerceRow>();
}

File: SMarketPlace/SMarketPlace.Web/Modules/Ecommerce/OrdenesEcommerce/OrdenesEcommerceRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace SMarketPlace.Ecommerce;

[LookupScript(Expiration = -1)]
[ConnectionKey("Default"), Module("Ecommerce"), TableName("OrdenesEcommerce")]
[DisplayName("Ordenes Ecommerce"), InstanceName("Ordenes Ecommerce")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
public sealed class OrdenesEcommerceRow : Row<OrdenesEcommerceRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Consecutivo"), Identity, IdProperty]
    public int? Consecutivo
    {
        get => fields.Consecutivo[this];
        set => fields.Consecutivo[this] = value;
    }

    [DisplayName("Id Pedido Ecommerce"), Column("idPedidoEcommerce"), Size(10), NotNull, QuickSearch, NameProperty]
    public string IdPedidoEcommerce
    {
        get => fields.IdPedidoEcommerce[this];
        set => fields.IdPedidoEcommerce[this] = value;
    }

    [DisplayName("Estado Pedido Internet"), Column("estado_pedido_internet"), Size(10), NotNull]
    public string EstadoPedidoInternet
    {
        get => fields.EstadoPedidoInternet[this];
        set => fields.EstadoPedidoInternet[this] = value;
    }

    public class RowFields : RowFieldsBase
    {
        public Int32Field Consecutivo;
        public StringField IdPedidoEcommerce;
        public StringField EstadoPedidoInternet;

    }
}